tensorboard/webapp/widgets/line_chart_v2/data_transformer.ts

```diff
@@ -0,0 +1,99 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {DataSeries} from './lib/public_types';
+
+/**
+ * Smoothes data series in y axis using smoothing algorithm from classical TensorBoard
+ * circa 2019-2020. 1st-order IIR low-pass filter to attenuate the higher-frequency
+ * components of the time-series.
+ * @param data DataSeries to smooth
+ * @param smoothingWeight Degree of smoothing. Number between 0 and 1, inclusive.
+ */
+export async function classicSmoothing(
+  data: DataSeries[],
+  smoothingWeight: number
+): Promise<Array<{srcId: string; points: DataSeries['points']}>> {
```

Comment:
Curious why we don’t return a `DataSeries[]` here? Is it useful to
rename the `id` field to `srcId`?

Not necessarily objecting; just wondering about the rationale.